ext/geos_c_impl/geometry.c

```diff
@@ -1158,6 +1158,46 @@ method_geometry_polygonize(VALUE self)
   return result;
 }
 
+VALUE
+primary_method_geometry_voronoi_diagram(VALUE _,
+                                        VALUE self,
+                                        VALUE env,
+                                        VALUE tolerance,
+                                        VALUE only_edges)
+{
+  VALUE result;
+  const RGeo_GeometryData* self_data;
+  const GEOSGeometry* self_geom;
+  GEOSGeometry* diagram;
+  const GEOSGeometry* env_geom = NULL;
+
+  Check_Type(tolerance, T_FLOAT);
+
+  result = Qnil;
+  self_data = RGEO_GEOMETRY_DATA_PTR(self);
+  self_geom = self_data->geom;
+
+  if (self_geom) {
+    if (RB_TEST(env)) {
+      env_geom = rgeo_convert_to_geos_geometry(self_data->factory, env, Qnil);
+      if (env_geom == NULL)
+        rb_raise(rb_eRGeoInvalidGeometry,
+                 "Could not use the given envelope with a GEOS factory. "
+                 "`RGeo::Feature.cast` failed.");
+    }
+    diagram = GEOSVoronoiDiagram(
+      self_geom, env_geom, rb_num2dbl(tolerance), RB_TEST(only_edges));
+
+    if (diagram == NULL) {
+      rb_raise(rb_eGeosError, "GEOS cannot create a voronoi diagram");
+    }
```

Comment:
should we remove this and let the handler that we're going to implement in the wrappers take care of raising?